### PR TITLE
[New Rule] PowerShell Script Block Logging Disabled

### LINKED ISSUE #1748: [New Rule] PowerShell Script Block Logging Disabled
## Description

Identifies the disable of PowerShell Script Block Logging. Attackers can disable the logging of PowerShell Scripts to
conceal their activities in the host.

## Example Data

Elastic Endpoint Security:

```
{
  "_index": ".ds-logs-endpoint.events.registry-default-2022.01.12-000001",
  "_type": "_doc",
  "_id": "-FDIsX4B7Kld_xqUnUCy",
  "_version": 1,
  "_score": 1,
  "_source": {
    "registry": {
      "hive": "HKLM",
      "path": "HKLM\\SOFTWARE\\Policies\\Microsoft\\Windows\\PowerShell\\ScriptBlockLogging\\EnableScriptBlockLogging",
      "data": {
        "strings": [
          "0"
        ],
        "type": "REG_DWORD"
      },
      "value": "EnableScriptBlockLogging",
      "key": "SOFTWARE\\Policies\\Microsoft\\Windows\\PowerShell\\ScriptBlockLogging"
    },
    "agent": {
      "id": "3832aecf-bfec-4ee4-9a5c-c257c05b578c",
      "type": "endpoint",
      "version": "7.16.2"
    },
    "process": {
      "Ext": {
        "ancestry": [
          "MzgzMmFlY2YtYmZlYy00ZWU0LTlhNWMtYzI1N2MwNWI1NzhjLTU2MzItMTMyODcxMTczMTAuMzg4MDc0NTAw",
          "MzgzMmFlY2YtYmZlYy00ZWU0LTlhNWMtYzI1N2MwNWI1NzhjLTU2MTYtMTMyODcxMTczMDkuNTM5MDEwMzAw",
          "MzgzMmFlY2YtYmZlYy00ZWU0LTlhNWMtYzI1N2MwNWI1NzhjLTU0OC0xMzI4NzExNzI2NS4xNzk0NDEwMDA="
        ]
      },
      "name": "powershell.exe",
      "pid": 13456,
      "entity_id": "MzgzMmFlY2YtYmZlYy00ZWU0LTlhNWMtYzI1N2MwNWI1NzhjLTEzNDU2LTEzMjg3NTE3Nzk2LjI3NzE0NzEwMA==",
      "executable": "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe"
    },
    "message": "Endpoint registry event",
    "@timestamp": "2022-02-01T01:16:57.9572808Z",
    "ecs": {
      "version": "1.11.0"
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "endpoint.events.registry"
    },
    "elastic": {
      "agent": {
        "id": "3832aecf-bfec-4ee4-9a5c-c257c05b578c"
      }
    },
    "host": {
      "hostname": "EXCC",
      "os": {
        "Ext": {
          "variant": "Windows 10 Enterprise Evaluation"
        },
        "kernel": "21H1 (10.0.19043.1466)",
        "name": "Windows",
        "family": "windows",
        "type": "windows",
        "version": "21H1 (10.0.19043.1466)",
        "platform": "windows",
        "full": "Windows 10 Enterprise Evaluation 21H1 (10.0.19043.1466)"
      },
      "ip": [
        "192.168.15.35"
      ],
      "name": "EXCC",
      "id": "cf593be5-65d6-4dfc-b8fd-ae790c3d8f93",
      "mac": [
        "08:00:27:ba:5e:64"
      ],
      "architecture": "x86_64"
    },
    "event": {
      "agent_id_status": "verified",
      "sequence": 719005,
      "ingested": "2022-01-31T20:17:01Z",
      "created": "2022-02-01T01:16:57.9572808Z",
      "kind": "event",
      "module": "endpoint",
      "action": "modification",
      "id": "MSX+plOd1q2Rp8HN++++chuk",
      "category": [
        "registry"
      ],
      "type": [
        "change"
      ],
      "dataset": "endpoint.events.registry"
    },
    "user": {
      "domain": "EXCC",
      "name": "Exchanger",
      "id": "S-1-5-21-1867456050-465470014-1116380518-1001"
    }
  },
  "fields": {
    "host.os.full.text": [
      "Windows 10 Enterprise Evaluation 21H1 (10.0.19043.1466)"
    ],
    "event.category": [
      "registry"
    ],
    "process.name.text": [
      "powershell.exe"
    ],
    "host.os.name.text": [
      "Windows"
    ],
    "host.os.full": [
      "Windows 10 Enterprise Evaluation 21H1 (10.0.19043.1466)"
    ],
    "host.hostname": [
      "EXCC"
    ],
    "process.pid": [
      13456
    ],
    "host.mac": [
      "08:00:27:ba:5e:64"
    ],
    "elastic.agent.id": [
      "3832aecf-bfec-4ee4-9a5c-c257c05b578c"
    ],
    "registry.hive": [
      "HKLM"
    ],
    "host.os.version": [
      "21H1 (10.0.19043.1466)"
    ],
    "host.os.name": [
      "Windows"
    ],
    "registry.key": [
      "SOFTWARE\\Policies\\Microsoft\\Windows\\PowerShell\\ScriptBlockLogging"
    ],
    "host.name": [
      "EXCC"
    ],
    "event.agent_id_status": [
      "verified"
    ],
    "event.kind": [
      "event"
    ],
    "registry.path": [
      "HKLM\\SOFTWARE\\Policies\\Microsoft\\Windows\\PowerShell\\ScriptBlockLogging\\EnableScriptBlockLogging"
    ],
    "user.id": [
      "S-1-5-21-1867456050-465470014-1116380518-1001"
    ],
    "host.os.type": [
      "windows"
    ],
    "process.Ext.ancestry": [
      "MzgzMmFlY2YtYmZlYy00ZWU0LTlhNWMtYzI1N2MwNWI1NzhjLTU2MzItMTMyODcxMTczMTAuMzg4MDc0NTAw",
      "MzgzMmFlY2YtYmZlYy00ZWU0LTlhNWMtYzI1N2MwNWI1NzhjLTU2MTYtMTMyODcxMTczMDkuNTM5MDEwMzAw",
      "MzgzMmFlY2YtYmZlYy00ZWU0LTlhNWMtYzI1N2MwNWI1NzhjLTU0OC0xMzI4NzExNzI2NS4xNzk0NDEwMDA="
    ],
    "registry.data.strings": [
      "0"
    ],
    "data_stream.type": [
      "logs"
    ],
    "host.architecture": [
      "x86_64"
    ],
    "process.name": [
      "powershell.exe"
    ],
    "agent.id": [
      "3832aecf-bfec-4ee4-9a5c-c257c05b578c"
    ],
    "ecs.version": [
      "1.11.0"
    ],
    "event.created": [
      "2022-02-01T01:16:57.957Z"
    ],
    "agent.version": [
      "7.16.2"
    ],
    "registry.data.type": [
      "REG_DWORD"
    ],
    "host.os.family": [
      "windows"
    ],
    "user.name": [
      "Exchanger"
    ],
    "process.entity_id": [
      "MzgzMmFlY2YtYmZlYy00ZWU0LTlhNWMtYzI1N2MwNWI1NzhjLTEzNDU2LTEzMjg3NTE3Nzk2LjI3NzE0NzEwMA=="
    ],
    "host.ip": [
      "192.168.15.35"
    ],
    "process.executable.caseless": [
      "c:\\windows\\system32\\windowspowershell\\v1.0\\powershell.exe"
    ],
    "event.sequence": [
      719005
    ],
    "agent.type": [
      "endpoint"
    ],
    "process.executable.text": [
      "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe"
    ],
    "event.module": [
      "endpoint"
    ],
    "host.os.kernel": [
      "21H1 (10.0.19043.1466)"
    ],
    "host.os.full.caseless": [
      "windows 10 enterprise evaluation 21h1 (10.0.19043.1466)"
    ],
    "user.domain": [
      "EXCC"
    ],
    "host.id": [
      "cf593be5-65d6-4dfc-b8fd-ae790c3d8f93"
    ],
    "process.name.caseless": [
      "powershell.exe"
    ],
    "process.executable": [
      "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe"
    ],
    "data_stream.namespace": [
      "default"
    ],
    "message": [
      "Endpoint registry event"
    ],
    "host.os.Ext.variant": [
      "Windows 10 Enterprise Evaluation"
    ],
    "event.ingested": [
      "2022-01-31T20:17:01.000Z"
    ],
    "event.action": [
      "modification"
    ],
    "@timestamp": [
      "2022-02-01T01:16:57.957Z"
    ],
    "host.os.platform": [
      "windows"
    ],
    "data_stream.dataset": [
      "endpoint.events.registry"
    ],
    "event.type": [
      "change"
    ],
    "event.id": [
      "MSX+plOd1q2Rp8HN++++chuk"
    ],
    "host.os.name.caseless": [
      "windows"
    ],
    "event.dataset": [
      "endpoint.events.registry"
    ],
    "user.name.text": [
      "Exchanger"
    ],
    "registry.value": [
      "EnableScriptBlockLogging"
    ]
  }
}
```

Sysmon:

```
{
  "_index": ".ds-logs-windows.sysmon_operational-default-2022.01.12-000001",
  "_type": "_doc",
  "_id": "J1J8sn4B7Kld_xqU2erY",
  "_version": 1,
  "_score": 1,
  "_source": {
    "registry": {
      "hive": "HKLM",
      "path": "HKLM\\SOFTWARE\\Policies\\Microsoft\\Windows\\PowerShell\\ScriptBlockLogging\\EnableScriptBlockLogging",
      "data": {
        "strings": [
          "0x00000000"
        ],
        "type": "SZ_DWORD"
      },
      "value": "EnableScriptBlockLogging",
      "key": "SOFTWARE\\Policies\\Microsoft\\Windows\\PowerShell\\ScriptBlockLogging\\EnableScriptBlockLogging"
    },
    "agent": {
      "hostname": "EXCC",
      "name": "EXCC",
      "id": "3832aecf-bfec-4ee4-9a5c-c257c05b578c",
      "ephemeral_id": "c741a80c-35f2-4bf0-a903-cd3503087cd0",
      "type": "filebeat",
      "version": "7.16.2"
    },
    "process": {
      "name": "powershell.exe",
      "pid": 13456,
      "entity_id": "{cf593be5-dd64-61ee-3910-000000000c00}",
      "executable": "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe"
    },
    "winlog": {
      "computer_name": "EXCC.exchangelab.com",
      "process": {
        "pid": 2456,
        "thread": {
          "id": 480
        }
      },
      "channel": "Microsoft-Windows-Sysmon/Operational",
      "event_data": {
        "EventType": "SetValue"
      },
      "opcode": "Info",
      "version": 2,
      "record_id": "743775",
      "event_id": "13",
      "task": "Registry value set (rule: RegistryEvent)",
      "provider_guid": "{5770385f-c22a-43e0-bf4c-06f5698ffbd9}",
      "api": "wineventlog",
      "provider_name": "Microsoft-Windows-Sysmon",
      "user": {
        "identifier": "S-1-5-18",
        "domain": "NT AUTHORITY",
        "name": "SYSTEM",
        "type": "User"
      }
    },
    "log": {
      "level": "information"
    },
    "elastic_agent": {
      "id": "3832aecf-bfec-4ee4-9a5c-c257c05b578c",
      "version": "7.16.2",
      "snapshot": false
    },
    "message": "Registry value set:\nRuleName: -\nEventType: SetValue\nUtcTime: 2022-02-01 04:33:59.612\nProcessGuid: {cf593be5-dd64-61ee-3910-000000000c00}\nProcessId: 13456\nImage: C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\nTargetObject: HKLM\\SOFTWARE\\Policies\\Microsoft\\Windows\\PowerShell\\ScriptBlockLogging\\EnableScriptBlockLogging\nDetails: DWORD (0x00000000)\nUser: EXCC\\Exchanger",
    "input": {
      "type": "winlog"
    },
    "@timestamp": "2022-02-01T04:33:59.612Z",
    "ecs": {
      "version": "1.12.0"
    },
    "related": {
      "user": [
        "Exchanger"
      ]
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "windows.sysmon_operational"
    },
    "host": {
      "hostname": "EXCC",
      "os": {
        "build": "19043.1466",
        "kernel": "10.0.19041.1466 (WinBuild.160101.0800)",
        "name": "Windows 10 Enterprise Evaluation",
        "family": "windows",
        "type": "windows",
        "version": "10.0",
        "platform": "windows"
      },
      "ip": [
        "10.0.3.15"
      ],
      "name": "EXCC.exchangelab.com",
      "id": "cf593be5-65d6-4dfc-b8fd-ae790c3d8f93",
      "mac": [
        "08:00:27:ba:5e:64"
      ],
      "architecture": "x86_64"
    },
    "event": {
      "agent_id_status": "verified",
      "ingested": "2022-01-31T23:33:53Z",
      "code": "13",
      "provider": "Microsoft-Windows-Sysmon",
      "created": "2022-02-01T04:34:00.811Z",
      "kind": "event",
      "action": "Registry value set (rule: RegistryEvent)",
      "category": [
        "configuration",
        "registry"
      ],
      "type": [
        "change"
      ],
      "dataset": "windows.sysmon_operational"
    },
    "user": {
      "domain": "EXCC",
      "name": "Exchanger",
      "id": "S-1-5-18"
    }
  },
  "fields": {
    "elastic_agent.version": [
      "7.16.2"
    ],
    "event.category": [
      "configuration",
      "registry"
    ],
    "host.os.name.text": [
      "Windows 10 Enterprise Evaluation"
    ],
    "winlog.provider_guid": [
      "{5770385f-c22a-43e0-bf4c-06f5698ffbd9}"
    ],
    "winlog.provider_name": [
      "Microsoft-Windows-Sysmon"
    ],
    "host.hostname": [
      "EXCC"
    ],
    "winlog.computer_name": [
      "EXCC.exchangelab.com"
    ],
    "process.pid": [
      13456
    ],
    "host.mac": [
      "08:00:27:ba:5e:64",
      "08:00:27:53:fa:ae"
    ],
    "winlog.process.pid": [
      2456
    ],
    "registry.hive": [
      "HKLM"
    ],
    "host.os.version": [
      "10.0"
    ],
    "winlog.record_id": [
      "743775"
    ],
    "host.os.name": [
      "Windows 10 Enterprise Evaluation"
    ],
    "registry.key": [
      "SOFTWARE\\Policies\\Microsoft\\Windows\\PowerShell\\ScriptBlockLogging\\EnableScriptBlockLogging"
    ],
    "log.level": [
      "information"
    ],
    "agent.name": [
      "EXCC"
    ],
    "host.name": [
      "EXCC.exchangelab.com"
    ],
    "event.agent_id_status": [
      "verified"
    ],
    "event.kind": [
      "event"
    ],
    "registry.path": [
      "HKLM\\SOFTWARE\\Policies\\Microsoft\\Windows\\PowerShell\\ScriptBlockLogging\\EnableScriptBlockLogging"
    ],
    "winlog.version": [
      2
    ],
    "host.os.type": [
      "windows"
    ],
    "user.id": [
      "S-1-5-18"
    ],
    "input.type": [
      "winlog"
    ],
    "registry.data.strings": [
      "0x00000000"
    ],
    "agent.hostname": [
      "EXCC"
    ],
    "data_stream.type": [
      "logs"
    ],
    "related.user": [
      "Exchanger"
    ],
    "host.architecture": [
      "x86_64"
    ],
    "process.name": [
      "powershell.exe"
    ],
    "event.provider": [
      "Microsoft-Windows-Sysmon"
    ],
    "event.code": [
      "13"
    ],
    "agent.id": [
      "3832aecf-bfec-4ee4-9a5c-c257c05b578c"
    ],
    "ecs.version": [
      "1.12.0"
    ],
    "event.created": [
      "2022-02-01T04:34:00.811Z"
    ],
    "agent.version": [
      "7.16.2"
    ],
    "registry.data.type": [
      "SZ_DWORD"
    ],
    "host.os.family": [
      "windows"
    ],
    "winlog.process.thread.id": [
      480
    ],
    "user.name": [
      "Exchanger"
    ],
    "process.entity_id": [
      "{cf593be5-dd64-61ee-3910-000000000c00}"
    ],
    "host.os.build": [
      "19043.1466"
    ],
    "winlog.user.type": [
      "User"
    ],
    "host.ip": [
      "10.0.3.15"
    ],
    "agent.type": [
      "filebeat"
    ],
    "event.module": [
      "windows"
    ],
    "host.os.kernel": [
      "10.0.19041.1466 (WinBuild.160101.0800)"
    ],
    "winlog.api": [
      "wineventlog"
    ],
    "elastic_agent.snapshot": [
      false
    ],
    "user.domain": [
      "EXCC"
    ],
    "host.id": [
      "cf593be5-65d6-4dfc-b8fd-ae790c3d8f93"
    ],
    "process.executable": [
      "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe"
    ],
    "winlog.user.identifier": [
      "S-1-5-18"
    ],
    "winlog.task": [
      "Registry value set (rule: RegistryEvent)"
    ],
    "winlog.user.domain": [
      "NT AUTHORITY"
    ],
    "elastic_agent.id": [
      "3832aecf-bfec-4ee4-9a5c-c257c05b578c"
    ],
    "data_stream.namespace": [
      "default"
    ],
    "message": [
      "Registry value set:\nRuleName: -\nEventType: SetValue\nUtcTime: 2022-02-01 04:33:59.612\nProcessGuid: {cf593be5-dd64-61ee-3910-000000000c00}\nProcessId: 13456\nImage: C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\nTargetObject: HKLM\\SOFTWARE\\Policies\\Microsoft\\Windows\\PowerShell\\ScriptBlockLogging\\EnableScriptBlockLogging\nDetails: DWORD (0x00000000)\nUser: EXCC\\Exchanger"
    ],
    "winlog.user.name": [
      "SYSTEM"
    ],
    "winlog.event_id": [
      "13"
    ],
    "event.ingested": [
      "2022-01-31T23:33:53.000Z"
    ],
    "event.action": [
      "Registry value set (rule: RegistryEvent)"
    ],
    "@timestamp": [
      "2022-02-01T04:33:59.612Z"
    ],
    "winlog.channel": [
      "Microsoft-Windows-Sysmon/Operational"
    ],
    "host.os.platform": [
      "windows"
    ],
    "winlog.event_data.EventType": [
      "SetValue"
    ],
    "data_stream.dataset": [
      "windows.sysmon_operational"
    ],
    "event.type": [
      "change"
    ],
    "winlog.opcode": [
      "Info"
    ],
    "agent.ephemeral_id": [
      "c741a80c-35f2-4bf0-a903-cd3503087cd0"
    ],
    "event.dataset": [
      "windows.sysmon_operational"
    ],
    "registry.value": [
      "EnableScriptBlockLogging"
    ]
  }
}
```

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issues

Resolves #1748 

## Summary

Identifies the disable of PowerShell Script Block Logging. Attackers can disable the logging of PowerShell Scripts to
conceal their activities in the host.

The rule works for both sysmon and Elastic Endpoint Security

### CODE REVIEW

File: rules/windows/defense_evasion_disable_posh_scriptblocklogging.toml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+[metadata]
+creation_date = "2022/01/31"
+maturity = "production"
+updated_date = "2022/01/31"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the disable of PowerShell Script Block Logging. Attackers can disable the logging of PowerShell Scripts to
+conceal their activities in the host.
```
Comment (Samirbous) [inline rules/windows/defense_evasion_disable_posh_scriptblocklogging.toml:10]:
```suggestion
Identifies attempt to disable PowerShell Script Block Logging via registry modification. Attackers may disable the logging of PowerShell Scripts to conceal their activities in the host.
```

**Review by Samirbous — APPROVED** [review]

File: rules/windows/defense_evasion_disable_posh_scriptblocklogging.toml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+[metadata]
+creation_date = "2022/01/31"
+maturity = "production"
+updated_date = "2022/01/31"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies attempt to disable PowerShell Script Block Logging via registry modification. Attackers may disable the
+logging of PowerShell Scripts to conceal their activities in the host.
```
Comment (brokensound77) [inline rules/windows/defense_evasion_disable_posh_scriptblocklogging.toml:10]:
```suggestion
Identifies attempts to disable PowerShell Script Block Logging via registry modification. Attackers may disable
PowerShell script logging to conceal their activities in the host and evade detection.
```

File: rules/windows/defense_evasion_disable_posh_scriptblocklogging.toml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+]
+risk_score = 47
+rule_id = "818e23e6-2094-4f0e-8c01-22d30f3506c6"
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+registry where event.type : "change" and
```
Comment (brokensound77) [inline rules/windows/defense_evasion_disable_posh_scriptblocklogging.toml:28]:
```suggestion
registry where event.type == "change" and
```

**Review by brokensound77 — APPROVED** [review]
left a few comments then LGTM 👍

**Pull request merged**